src/Bot.hs

```diff
@@ -40,9 +45,14 @@ app' = do
 
 typeRequest :: Text -> TypeBot ()
 typeRequest f = do
-  result <- hoogle f
+  (AppConfig _ s) <- lift ask
```

Comment:
Minor nit, but: I try to avoid pattern matching outside of where the data type is defined, to be more flexible to moving fields around later without have to update a ton of places. In this case an `appSearchEngine` accessor would be useful and I'd uses `s <- lift asks appSearchEngine` here. Similarly with `appConfig` at line 93.